sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/ColumnDefinition.scala

```diff
@@ -176,6 +177,18 @@ object ColumnDefinition {
           }
         }
 
+      case cmd: AlterColumns if cmd.specs.exists(_.newDefaultExpression.isDefined) =>
+        // Wrap analysis errors for default values in a more user-friendly message.
```

Comment:
not related to this PR but we can probably do the same error message improvement for `V2CreateTablePlan`